### PR TITLE
api.rst: document onedata/rest-cli usage.

### PR DESCRIPTION
**Pull request by gwarf**
Document API usage with some real examples.
WIP.

### CODE REVIEW

File: source/api.rst [hunk 1]
```diff
@@ -18,6 +18,8 @@ Tokens have to be generated from the **EGI DataHub** (Onezone) interface as
 documented in :ref:`auth-token-using-web-interface` or using a command line
 call as documented hereafter.
 
+Bear in mind that a unique API tokens can be used with both Onezone and Oneprovider.
```
Comment (enolfc) [inline source/api.rst]: s/tokens/token/

Comment (enolfc) [reply source/api.rst]: Also I would rather use "single" instead of "unique"

File: source/api.rst [hunk 2]
```diff
@@ -38,51 +40,119 @@ Testing the API with the REST client
...
 * ``ONEZONE_HOST``: name or IP of the Onezone host (to use Onezone API).
+* ``ONEPROVIDER_HOST``: name or IP of the Oneprovider host (to use Oneprovider API)
 
 .. code-block:: console
 
    # Getting the IDs of the available Handle Services
    curl -sS --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
-     -X GET \
-     "https://datahub.egi.eu/api/v3/onezone/user/handle_services"
+     "$HONEZONE_HOST/api/v3/onezone/user/handle_services"
```
Comment (enolfc) [inline source/api.rst]: s/HONEZONE/ONEZONE/

**Review by enolfc — CHANGES_REQUESTED** [review]
some typos here and there... thanks!

File: source/api.rst (hunk 1 above)
Comment (gwarf) [reply source/api.rst]: Agreed.

File: source/api.rst (hunk 2 above)
Comment (gwarf) [reply source/api.rst]: Fixed.

**Comment by gwarf** [comment]
@enolfc , your turn :)

**Review by enolfc — APPROVED** [review]
Looks good!

**Pull request merged**